### PR TITLE
doc: release-notes/3.3: adds bits on PCIE, UART, IPM, and I3C

### PR DESCRIPTION
**Pull request by dcpleung**
This adds a few bits in the release notes for 3.3 on PCIE, UART, IPM and I3C.

### CODE REVIEW

**Review by stephanosio — DISMISSED** [review]

**Comment by stephanosio** [comment]
@dcpleung This needs a rebase.

**Comment by dcpleung** [comment]
> @dcpleung This needs a rebase.

Rebased.

**Review by stephanosio — DISMISSED** [review]

File: doc/releases/release-notes-3.3.rst [hunk 1]
```diff
@@ -750,6 +761,19 @@ Drivers and Sensors
...
+
+  * Added new API :c:func:`pcie_scan` to scan for devices.
+
+    * This interates through the the buses and devices which are expected to
+      exist. The old method was to try all possible combination of buses
+      and devices to determine if there is a device there.
+      :c:func:`pci_init` and :c:func:`pcie_bdf_lookup` have been updated to
+      use this new API.
+
+    * :c:func:`pcie_scan` also introduces a callback mechanism for when
```
Comment (nordicjm) [inline doc/releases/release-notes-3.3.rst:774]: None of these link to the functions

File: doc/releases/release-notes-3.3.rst [hunk 2]
```diff
@@ -732,6 +732,14 @@ Drivers and Sensors
 
 * I3C
 
+  * Added a new target device API :c:func:`i3c_target_tx_write` to
```
[the remark this answers is not in the document]
Comment (dcpleung) [reply doc/releases/release-notes-3.3.rst:752]: Added the necessary doxygen group to the I3C page. This should link to that.

File: doc/releases/release-notes-3.3.rst [hunk 3]
```diff
@@ -750,6 +761,19 @@ Drivers and Sensors
 
 * PCIE
 
+  * Support for accessing I/O BARs, which was previously removed, is back.
+
+  * Added new API :c:func:`pcie_scan` to scan for devices.
```
[the remark this answers is not in the document]
Comment (dcpleung) [reply doc/releases/release-notes-3.3.rst:783]: Currently there is no PCI-E page which includes the necessary doxygen group, hence linking is of no effect. A new PCI-E page will need to be added in the future.

File: doc/releases/release-notes-3.3.rst (hunk 1 above)
Comment (dcpleung) [reply doc/releases/release-notes-3.3.rst:791]: (ditto)

**Review by nordicjm — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Pull request merged**